Common/DiagnosticAssemblyScanner.cs

```diff
@@ -32,41 +32,62 @@ public DiagnosticAssemblyScanner(ILogger logger, params string[] additionalSearc
         }
 
         /// <summary>
+        /// Loads all assemblies in the given directory and instantiates Roslyn diagnostic objects - i.e. existing types deriving from
+        /// <see cref="DiagnosticAnalyzer"/>
+        /// </summary>
+        /// <returns>empty enumerable if no diagnostics were found</returns>
+        public IEnumerable<DiagnosticAnalyzer> InstantiateDiagnosticsFromDirectory(string directoryPath, string language)
+        {
+            List<DiagnosticAnalyzer> analyzers = new List<DiagnosticAnalyzer>();
+
+            foreach (string assemblyPath in Directory.GetFiles(directoryPath, "*.dll", SearchOption.AllDirectories))
+            {
+                analyzers.AddRange(InstantiateDiagnosticsFromAssembly(assemblyPath, language));
+            }
+            
+            if (analyzers.Any())
+            {
+                return analyzers;
+            }
+            return Enumerable.Empty<DiagnosticAnalyzer>();
+        }
+
+        /// <summary>
         /// Loads the given assembly and instantiates Roslyn diagnostic objects - i.e. existing types deriving from
         /// <see cref="DiagnosticAnalyzer"/>
         /// </summary>
         /// <returns>empty enumerable if no diagnostics were found</returns>
         public IEnumerable<DiagnosticAnalyzer> InstantiateDiagnosticsFromAssembly(string assemblyPath, string language)
```

Comment:
Forgot to make a note of this yesterday. It proved to be very difficult to test loading assemblies from individual folders, so the assembly-level loading is still the point of entry for testing. I felt that (given that folder-level assembly loading is a foreach and AddRange), this is still acceptable.

There should probably be a for-test comment here.